src/dat/controllers/CustomController.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * dat-gui JavaScript Controller Library
+ * http://code.google.com/p/dat-gui
+ *
+ * Copyright 2011 Data Arts Team, Google Creative Lab
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+import Controller from './Controller';
+
+/**
+ * @class Represents a custom controller.
+ * @param {init} callback function for adding of elements into this.domElement
+ */
+class CustomController extends Controller{
+    constructor(init) {
+    super({});
+
+    init( this );
+    this.custom = true;
+  }
+}
+
```

Comment:
From [your example](https://raw.githack.com/anhr/three.js/dev/examples/webgl_custom_controller.html) –

```js
var controllerPlayRate1 = folder1.add( new dat.controllers.CustomController( function ( controller ) {

  controllerPlay( controller, buttons1 );

} ),
{

  playRate: 1,

}, 'playRate', 1, 25, 1 ).onChange( function ( value ) {

  onChangePlayRate( value );

} );
```

I'd rather have an API where custom controllers can be used more similarly to the builtin controller types. Users would not instantiate CustomController directly (i.e. it's an "abstract" class) but would instantiate subtypes. For example, they should do this:

```js
class KnobController extends CustomController {
  constructor ( object, name, ...opts ) {
    super( object, name );
    // ... set up options if needed
  }
}

const api = {
  color: '#ffffff',
  value: 0.5
};

const gui = new dat.GUI();
gui.add( api, 'color' );
gui.add( new KnobController( api, 'value' ) );
```

I believe that can be implemented without requiring an `init` callback, and without needing a `addCustomController` method (since `add` works just as well).

Reply:
Please see my new version of the [CustomController.js](https://github.com/anhr/dat.gui/blob/CustomController/src/dat/controllers/CustomController.js) as you asked me. I have rewired my own [ES module](https://github.com/anhr/customController) for testing and use it in my [example](https://raw.githack.com/anhr/three.js/dev/examples/webgl_custom_controller.html).